src/store/modalsSlice.ts

```diff
@@ -28,12 +28,19 @@ const modalsSlice = createSlice({
   name: "modals",
   initialState,
   reducers: {
-    handleVisibility(state, action: PayloadAction<ModalsState>) {
+    handleModalVisibilityAction(state, action: PayloadAction<ModalsState>) {
       const found = state.find((item) => item.id === action.payload.id);
-      if (found) found.isOpen = action.payload.isOpen;
+      if (found) {
+        found.isOpen = action.payload.isOpen;
+      } else {
+        state.push({
+          id: action.payload.id,
+          isOpen: action.payload.isOpen,
+        });
+      }
```

Comment:
Code is okay that way but I don't really like 'else' keyword, see if it's more readable: 
```tsx
if (found) {
found.isOpen = action.payload.isOpen
return
}

state.push({
          id: action.payload.id,
          isOpen: action.payload.isOpen,
        });
       ```